src/Microsoft.Azure.SignalR.Protocols/CheckWithAckMessage.cs

```diff
@@ -0,0 +1,122 @@
+    {
+        /// <summary>
+        /// Gets or sets the user Id.
+        /// </summary>
+        public string UserId { get; set; }
+
+        /// <summary>
+        /// Gets or sets the group name.
+        /// </summary>
+        public string GroupName { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LeaveGroupWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="userId">The connection Id.</param>
+        /// <param name="groupName">The group name, from which the connection will leave.</param>
+        /// <param name="ackId">The ack Id</param>
+        /// <param name="tracingId">The tracing Id of the message.</param>
+        public CheckUserInGroupWithAckMessage(string userId, string groupName, int ackId = 0, ulong? tracingId = null) : base(ackId, tracingId)
+        {
+            UserId = userId;
+            GroupName = groupName;
+        }
+    }
+
+    /// <summary>
+    /// A waiting for ack check-any-connection-in-group message.
+    /// </summary>
+    public class CheckAnyConnectionInGroupWithAckMessage : CheckWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the group name.
+        /// </summary>
+        public string GroupName { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CheckAnyConnectionInGroupWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="groupName">The group name, from which the connection will leave.</param>
+        /// <param name="ackId">The ack Id</param>
+        /// <param name="tracingId">The tracing Id of the message.</param>
+        public CheckAnyConnectionInGroupWithAckMessage(string groupName, int ackId = 0, ulong? tracingId = null) : base(ackId, tracingId)
+        {
+            GroupName = groupName;
+        }
+    }
+
+    /// <summary>
+    /// A waiting for ack check-connection-existence message.
+    /// </summary>
+    public class CheckConnectionExistenceWithAckMessage : CheckWithAckMessage
+    {
+        /// <summary>
+        /// Gets or sets the connection Id.
+        /// </summary>
+        public string ConnectionId { get; set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CheckConnectionExistenceWithAckMessage"/> class.
+        /// </summary>
+        /// <param name="connectionId">The connection Id.</param>
+        /// <param name="ackId">The ack Id</param>
+        /// <param name="tracingId">The tracing Id of the message.</param>
+        public CheckConnectionExistenceWithAckMessage(string connectionId, int ackId = 0, ulong? tracingId = null) : base(ackId, tracingId)
+        {
+            ConnectionId = connectionId;
+        }
+    }
+
+    /// <summary>
+    /// A waiting for ack check-connection-existence-as-user message.
+    /// </summary>
+    public class CheckAnyConnectionInUserWithAckMessage : CheckWithAckMessage
```

Comment:
`InUser` -> `ForUser` ?

Reply:
or `OfUser`

Reply:
I still think `in` is better,
* Check Any Connection **In** User
* Check Any Connection **Of** User
* Check Any Connection **For** User
What do you think? @KKhurin @vicancy

Reply:
it's out of date